ui/src/app/store/domain/slice.ts

```diff
@@ -162,6 +164,66 @@ const domainSlice = createSlice({
     ) => {
       state.active = action.payload?.id || null;
     },
+    createRecord: {
+      prepare: (
```

Comment:
We usually put the action param types in the types dir so that we can reuse them elsewhere:

https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/domain/types/actions.ts

This will require passing in the params as an object.